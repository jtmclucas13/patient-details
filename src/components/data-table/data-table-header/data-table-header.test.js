import React from 'react';
import { shallow } from 'enzyme';
import DataTableHeader from './data-table-header';

describe('component DataTableHeader', () => {
    test('renders headers from columns', () => {
        const columns = [{ displayName: 'Column 1', key: '123' }];
        const wrapper = shallowRenderDataTableHeader({ columns });

        expect(wrapper).toMatchSnapshot();
    });

    test('renders empty header if no columns given', () => {
        const wrapper = shallowRenderDataTableHeader();

        expect(wrapper).toMatchSnapshot();
    });
});

/**
* Helper function for rendering DataTableHeader in tests without boilerplate required props
* @param {Object} [props={}] props for DataTableHeader
* @return {ShallowWrapper} enzyme wrapper for DataTableHeader 
*/
function shallowRenderDataTableHeader(props = {}) {
    const propsToUse = {
        columns: [],
        ...props,
    };

    return shallow(<DataTableHeader {...propsToUse} />);
}
