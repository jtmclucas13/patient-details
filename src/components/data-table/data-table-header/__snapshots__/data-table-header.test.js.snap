// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component DataTableHeader renders empty header if no columns given 1`] = `
<thead
  className="data-table-header"
>
  <tr />
</thead>
`;

exports[`component DataTableHeader renders headers from columns 1`] = `
<thead
  className="data-table-header"
>
  <tr>
    <th
      className="data-table-header__cell"
      key="123"
    >
      Column 1
    </th>
  </tr>
</thead>
`;
